extensions/Worker.Extensions.ServiceBus/src/ServiceBusReceivedMessageConverter.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Messaging.ServiceBus;
+using Azure.Messaging.ServiceBus.Primitives;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Microsoft.Azure.Functions.Worker.Core;
+
+namespace Microsoft.Azure.Functions.Worker;
+
+internal class ServiceBusReceivedMessageConverter : IInputConverter
+{
+    public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+    {
+        ConversionResult result = context?.Source switch
+        {
+            ModelBindingData binding => ConversionResult.Success(ConvertToServiceBusReceivedMessage(binding)),
+            CollectionModelBindingData collection when context.TargetType.IsArray => ConversionResult.Success(collection.ModelBindingDataArray.Select(ConvertToServiceBusReceivedMessage).ToArray()),
```

Comment:
I copied the IsArray logic from the Blobs PR - https://github.com/Azure/azure-functions-dotnet-worker/pull/1108/files#diff-f26f3ef6a62a761265eb94f5eccbbb343516e36211dcb1a96074ca4b4063d947R95